mock/mock.go

```diff
@@ -584,6 +601,14 @@ func (m *Mock) MethodCalled(methodName string, arguments ...interface{}) Argumen
 	returnArgs := call.ReturnArguments
 	m.mutex.Unlock()
 
+	m.mutex.Lock()
+	returnFn := call.returnFn
+	m.mutex.Unlock()
+
+	if returnFn != nil {
```

Comment:
Because call.ReturnArguments is exported it should probably be checked by this condition rather than returnFn, eg:
```go
	m = &myMock{}
	m.On("Do").Return("one")
	calls := m.Calls
	m.On("Do").Return("one").Unset()
	m.On("Do").ReturnFn(func(args mock.Arguments) mock.Arguments { return mock.Arguments{"two"} })
	m.Calls = calls
	assert.Equal(t, "one", m.Do())
```

ie. Call.ReturnArguments should always override Call.returnFn